File: sdks/ts/src/sdk/models/shared/fileversions.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { FileVersion } from "./fileversion";


export enum FileVersionsOrderDirectionEnum {
    Asc = "ASC",
    Desc = "DESC"
}


// FileVersionsOrder
/** 
 * The order in which a pagination is ordered
**/
export class FileVersionsOrder extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=by" })
  by?: string;

  @SpeakeasyMetadata({ data: "json, name=direction" })
  direction?: FileVersionsOrderDirectionEnum;
}


// FileVersions
/** 
 * The part of an API response that describes pagination
**/
export class FileVersions extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: FileVersion })
  entries?: FileVersion[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=offset" })
  offset?: number;

  @SpeakeasyMetadata({ data: "json, name=order", elemType: FileVersionsOrder })
  order?: FileVersionsOrder[];

  @SpeakeasyMetadata({ data: "json, name=total_count" })
  totalCount?: number;
}

File: java-sdk/lib/src/main/java/models/shared/FileVersions.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * FileVersions
 * The part of an API response that describes pagination
**/
public class FileVersions {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("entries")
    public FileVersion[] entries;
    public FileVersions withEntries(FileVersion[] entries) {
        this.entries = entries;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("limit")
    public Long limit;
    public FileVersions withLimit(Long limit) {
        this.limit = limit;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("offset")
    public Long offset;
    public FileVersions withOffset(Long offset) {
        this.offset = offset;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("order")
    public FileVersionsOrder[] order;
    public FileVersions withOrder(FileVersionsOrder[] order) {
        this.order = order;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("total_count")
    public Long totalCount;
    public FileVersions withTotalCount(Long totalCount) {
        this.totalCount = totalCount;
        return this;
    }
}